crates/sui-indexer-alt-jsonrpc/src/api/write.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use fastcrypto::encoding::Base64;
+use jsonrpsee::{core::RpcResult, http_client::HttpClient, proc_macros::rpc};
+use sui_json_rpc_types::{
+    DryRunTransactionBlockResponse, SuiTransactionBlockResponse, SuiTransactionBlockResponseOptions,
+};
+use sui_open_rpc::Module;
+use sui_open_rpc_macros::open_rpc;
+use sui_types::quorum_driver_types::ExecuteTransactionRequestType;
+
+use crate::error::internal_error_object;
+
+use super::rpc_module::RpcModule;
+
+#[open_rpc(namespace = "sui", tag = "Write API")]
+#[rpc(server, client, namespace = "sui")]
+pub trait WriteApi {
+    /// Execute the transaction and wait for results if desired.
+    /// Request types:
+    /// 1. WaitForEffectsCert: waits for TransactionEffectsCert and then return to client.
+    ///     This mode is a proxy for transaction finality.
+    /// 2. WaitForLocalExecution: waits for TransactionEffectsCert and make sure the node
+    ///     executed the transaction locally before returning the client. The local execution
+    ///     makes sure this node is aware of this transaction when client fires subsequent queries.
+    ///     However if the node fails to execute the transaction locally in a timely manner,
+    ///     a bool type in the response is set to false to indicated the case.
```

Comment:
What's the latest on deprecating `WaitForLocalExecution`? Should we be supporting different request types? Does `WaitForLocalExecution` still have meaning for us, given "local" means the fullnode, and not this RPC service? We can no longer guarantee that the RPC will return a response that is consistent with the effects of this transaction, after `executeTransactionBlock(..., WaitForLocalExecution)` returns.

Reply:
I've consulted Brandon who told me that this mode is still in the process of being deprecated but technically usable, and that for new services like rpc alt we should disable this mode. I will update the PR to return an error if user uses this mode.